tools/kth_threshold.cpp

```diff
@@ -0,0 +1,212 @@
+#include <iostream>
+#include <optional>
+#include <unordered_set>
+
+#include "boost/algorithm/string/classification.hpp"
+#include "boost/algorithm/string/split.hpp"
+#include <boost/functional/hash.hpp>
+
+#include "mio/mmap.hpp"
+#include "spdlog/sinks/stdout_color_sinks.h"
+#include "spdlog/spdlog.h"
+
+#include "mappable/mapper.hpp"
+
+#include "app.hpp"
+#include "cursor/max_scored_cursor.hpp"
+#include "index_types.hpp"
+#include "io.hpp"
+#include "query/algorithm.hpp"
+#include "util/util.hpp"
+#include "wand_data_compressed.hpp"
+#include "wand_data_raw.hpp"
+
+#include "query/algorithm.hpp"
+#include "scorer/scorer.hpp"
+
+#include "CLI/CLI.hpp"
+
+using namespace pisa;
+
+std::set<uint32_t> parse_tuple(std::string const& line, size_t k)
+{
+    std::vector<std::string> term_ids;
+    boost::algorithm::split(term_ids, line, boost::is_any_of(" \t"));
+    if (term_ids.size() != k) {
+        throw std::runtime_error(fmt::format(
+            "Wrong number of terms in line: {} (expected {} but found {})", line, k, term_ids.size()));
+    }
+
+    std::set<uint32_t> term_ids_int;
+    for (auto&& term_id: term_ids) {
+        try {
+            term_ids_int.insert(std::stoi(term_id));
+        } catch (const std::invalid_argument& e) {
```

Comment:
Since we're doing the same thing in both and there's no confusion where the potential exception is coming from, maybe we can just do `catch (const std::exception&) {` and  have one catch block.